lib/index.d.ts

```diff
@@ -5,31 +5,31 @@ export type CollectionConfig = {
     /**
      * - One or more glob patterns to match files to a collection
      */
-    pattern: string | string[];
+    pattern?: string | string[] | null;
```

Comment:
The `null`s are because the default options use them: https://github.com/metalsmith/collections/blob/362c20c6814b626850a41f8d245cde81f5af66df/src/index.js#L38